spec/package-manager-spec.coffee

```diff
@@ -1145,44 +1165,60 @@ describe "PackageManager", ->
         expect(atom.packages.isPackageActive("package-that-throws-on-activate")).toBeTruthy()
         spyOn(badPack.mainModule, 'deactivate').andCallThrough()
 
+      waitsForPromise ->
         atom.packages.deactivatePackage("package-that-throws-on-activate")
+
+      runs ->
         expect(badPack.mainModule.deactivate).not.toHaveBeenCalled()
         expect(atom.packages.isPackageActive("package-that-throws-on-activate")).toBeFalsy()
 
     it "absorbs exceptions that are thrown by the package module's deactivate method", ->
       spyOn(console, 'error')
+      thrownError = null
 
       waitsForPromise ->
         atom.packages.activatePackage("package-that-throws-on-deactivate")
 
+      waitsForPromise ->
+        try
+          atom.packages.deactivatePackage("package-that-throws-on-deactivate")
+        catch error
+          thrownError = error
```

Comment:
I don't think this `catch` clause will ever be triggered. What you probably want is:

```coffee
waitsForPromise ->
  atom.packages.deactivatePackage("package-that-throws-on-deactivate").catch(e -> error = e)
```